miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java

```diff
@@ -388,6 +395,55 @@ public void deleteQc(Library library, Long qcId) throws IOException {
   }
 
   /**
+   * Turns indices into strings for easier comparison and changelog message concatentation.
+   * 
+   * @param indices
+   * @return
+   */
+  private Set<String> stringifyIndices(List<Index> indices) {
+    Set<String> original = new HashSet<>();
+    for (Index index : indices) {
+      if (index != null) {
+        original.add(index.getFamily().getName() + " - " + index.getLabel());
+      }
+    }
+    return original;
+  }
+
+  /**
+   * Create a changelog if the indices have changed.
+   * 
+   * @param originalIndices
+   * @param updatedIndices
+   * @param target
+   * @throws IOException
+   */
+  private void makeChangeLogForIndices(List<Index> originalIndices, List<Index> updatedIndices, Library target) throws IOException {
+    
+    Set<String> original = stringifyIndices(originalIndices);
+    Set<String> updated = stringifyIndices(updatedIndices);
+    Set<String> added = new TreeSet<>(updated);
+    added.removeAll(originalIndices);
+    Set<String> removed = new TreeSet<>(original);
+    removed.removeAll(updatedIndices);
+
+    if (!added.isEmpty() || !removed.isEmpty()) {
+      StringBuilder message = new StringBuilder();
+      message.append("Indices");
+      LimsUtils.appendSet(message, removed, "removed");
+      LimsUtils.appendSet(message, added, " Added");
```

Comment:
This will result in
* when removing only:
`Indices removed: Family - Index 01 (AAAAAA), Family - Index 02 (CCCCCC)`
* when adding only:
`Indices  Added: Family - Index 01 (AAAAAA), Family - Index 02 (CCCCCC)`
* when adding and removing:
`Indices removed: Family - Index 01 (AAAAAA), Family - Index 02 (CCCCCC) Added: Family - Index 03 (GGGGGG), Family - Index 04 (TTTTTT)`

It's not visible here, but there's an extra space before "Added" in both cases since `LimsUtils#appendSet` also adds a space. The space here should be removed. "Added" should match "removed" (no capital), and a semicolon would be nice to separate removals from additions in cases where both happen